services/documentation_service.py

```diff
@@ -1,44 +1,54 @@
 import asyncio
-from typing import Coroutine, List, Any
+import http.client
+from typing import Coroutine, List, Any, Dict
 from schemas.documentation_generation import DocsStatusEnum, FirestoreDocumentationCreateModel, \
     FirestoreDocumentationUpdateModel, GeneratedDocResponse, LlmModelEnum
 from services.clients.firebase_client import FirebaseClient, get_firebase_client
 from google.cloud.firestore_v1 import DocumentReference
-from fastapi import BackgroundTasks
+from fastapi import BackgroundTasks, HTTPException
 
 from dotenv import load_dotenv
 
 from services.clients.llm_client import LLMClient
 from services.clients.openai_client import get_openai_client
 from services.clients.github_client import GitHubClient, get_github_client
 
+SYSTEM_PROMPT_FOR_FILES = """
+Your job is to provide very high-level documentation of code provided to you.
+    
+You will respond in Markdown format, with the following sections:
+## Description: (a string less than 100 words)
+## Insights: ([string, string, string])
+"""
+
 
 class DocumentationService:
     def __init__(self, llm_client: LLMClient, github_client: GitHubClient, firebase_client: FirebaseClient):
         self.llm_client = llm_client
         self.github_client = github_client
         self.firebase_client = firebase_client
-        self.system_prompt_for_file = """
-            Your job is to provide very high-level documentation of code provided to you.
-    
-            You will respond in Markdown format, with the following sections:
-            ## Description: (a string less than 100 words)
-            ## Insights: ([string, string, string])
-            
-            Here is the code:
-            """
-
-    async def generate_doc_for_github_file(self, file_url: str):
+        self.system_prompt_for_file = SYSTEM_PROMPT_FOR_FILES
+
+    async def generate_doc_for_github_file(
+            self,
+            file_url: str,
+            model: LlmModelEnum = LlmModelEnum.MIXTRAL
+    ):
         if not file_url:
             raise ValueError("File url cannot be empty")
 
         file_content = self.github_client.read_file(file_url)
 
+        user_prompt = "Here is the code: \n" + file_content
+
         llm_response = await self.llm_client.generate_text(
-            prompt=file_content,
+            model=model,
+            prompt=user_prompt,
             system_prompt=self.system_prompt_for_file,
             max_tokens=500
         )
+        if llm_response[0] == " ":
+            llm_response = llm_response[1:]
```

Comment:
Idk if it would be cleaner to have this in a helper function? 
then have `llm_response = function()`